node/actors/network/src/metrics.rs

```diff
@@ -245,58 +253,58 @@ impl MeteredStreamStats {
             received: 0.into(),
             established: SystemTime::now(),
             peer_addr,
-            sent_buffer: VecDeque::new(),
-            received_buffer: VecDeque::new(),
+            current_minute_sent: 0.into(),
+            previous_minute_sent: 0.into(),
+            current_minute_received: 0.into(),
+            previous_minute_received: 0.into(),
+            current_minute_start: 0.into(),
         }
     }
 
-    fn read(&mut self, amount: u64) {
+    fn read(&self, amount: u64) {
+        self.update_minute();
         self.received.fetch_add(amount, Ordering::Relaxed);
-        self.received_buffer.push_back((Instant::now(), amount));
-        self.cleanup_buffers();
+        self.current_minute_received
+            .fetch_add(amount, Ordering::Relaxed);
     }
 
-    fn wrote(&mut self, amount: u64) {
+    fn wrote(&self, amount: u64) {
+        self.update_minute();
         self.sent.fetch_add(amount, Ordering::Relaxed);
-        self.sent_buffer.push_back((Instant::now(), amount));
-        self.cleanup_buffers();
+        self.current_minute_sent
+            .fetch_add(amount, Ordering::Relaxed);
     }
 
-    fn cleanup_buffers(&mut self) {
-        let one_minute_ago = Instant::now() - Duration::from_secs(60);
+    fn update_minute(&self) {
+        let now = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .expect("Time went backwards")
```

Comment:
well this would mean that we went back before 1970 XD

Reply:
So the error message is indeed correct!